src/main/scala/core/scraper/Aggregator.scala

```diff
@@ -75,6 +75,8 @@ object Aggregator:
               val1
             case (_, Some(val2)) =>
               val2
+            case _ =>
+              throw new IllegalStateException("An unexpected error occurs.")
```

Comment:
Is it really necessary using Exceptions here?

Reply:
I agree, there should probably be a different design

Reply:
By means of unchecked tag I avoided the usage of the exception